data-runtime/src/main/java/io/micronaut/data/runtime/intercept/AbstractQueryInterceptor.java

```diff
@@ -434,30 +436,63 @@ protected <RT> Optional<RT> getParameterInRole(MethodInvocationContext<?, ?> con
      */
     @NonNull
     protected Pageable getPageable(MethodInvocationContext<?, ?> context) {
-        Pageable pageable = getParameterInRole(context, TypeRole.PAGEABLE, Pageable.class).orElse(null);
+        Pageable pageable = getPageableInRole(context);
         if (pageable == null) {
```

Comment:
This method is confusing, we never store direct page index / size into metadata. Offset is never using in the implementation